src/ImageSharp/Processing/Processors/Quantization/EuclideanPixelMap{TPixel}.cs

```diff
@@ -91,16 +105,43 @@ public void Clear(ReadOnlyMemory<TPixel> palette)
         this.cache.Clear();
     }
 
+    /// <summary>
+    /// Allows setting the transparent index after construction.
+    /// </summary>
+    /// <param name="index">An explicit index at which to match transparent pixels.</param>
+    public void SetTransparentIndex(int index) => this.transparentIndex = index;
+
     [MethodImpl(InliningOptions.ShortMethod)]
     private int GetClosestColorSlow(Rgba32 rgba, ref TPixel paletteRef, out TPixel match)
     {
         // Loop through the palette and find the nearest match.
         int index = 0;
         float leastDistance = float.MaxValue;
+
+        if (this.transparentIndex >= 0 && rgba == default)
+        {
+            // We have explicit instructions. No need to search.
+            index = this.transparentIndex;
+            this.cache.Add(rgba, (byte)index);
+
+            if (index >= 0 && index < this.Palette.Length)
+            {
+                match = Unsafe.Add(ref paletteRef, (uint)index);
+            }
+            else
+            {
+                Unsafe.SkipInit(out TPixel pixel);
+                pixel.FromScaledVector4(Vector4.Zero);
+                match = pixel;
+            }
+
+            return index;
+        }
```

Comment:
This is now penalizing all quantizers and decoders while only gif is using `transparentIndex`. Can't we  keep the transparency detection logic in `GifDecoderCore`?

Reply:
I couldn't figure out a way to do it. Given that the slow code is actually the loop through the palette I thought this addition check outside the loop would be ok. 

I actually think Bmp will end up using it eventually when we finally expose a decoded palette there.

Reply:
"Slow" actually means we run it every time a new value is added to `ColorDistanceCache` which should be still quite common for an image with a wide original palette. I also don't see an easy way to optimize this, we should probably introduce multiple variants of `EuclideanPixelMap<TPixel>` and `PalettQuantizer<TPixel>` in the future.

Reply:
I've been thinking about this and compared to the actual iteration the `>=0` check should be barely detectable. The comparison operator won't even kick in.

Reply:
The comparison is indeed cheap, the problem is not with that. We have a lot of code added to the method  which may lead to more frequent L1 instruction cache misses. Branches can also hurt CPU pipelining.

Since `GetQuantizedColor` is being called per-pixel, the method should be as slim as possible, ideally containing no branching except the pixelmap lookup. We should be able to achieve this (and greatly reduce the memory footprint for pixel types without an alpha!) by separating different `OctreeQuantizer<T>` setups to separate classes.

If we could look into this together with optimization opportunities in `ErrorDitherer`, we could achieve significant improvements in the quantization space. Not a small amount of work though.

Reply:
Yeah true. I'll not touch this until you have a chance to make changes. We can look at greater optimization opportunities for v4